### ISSUE TITLE
[404] /qwik-prefetch-service-worker.js

### ISSUE THREAD

**Issue by Hunnnn77** [QwikDev/astro#55]
I'm running output: server in astro, i can't understand why this 404 message comes up in log?
luckly, there's no problems running app realated this message.
```
"@builder.io/qwik": "^1.3.5"
"@qwikdev/astro": "^0.4.0"
"astro": "^4.1.2",

14:40:20 [200] / 4ms
14:40:20 [200] /api/0 0ms
14:40:20 [404] /qwik-prefetch-service-worker.js
```

- Error at browser
```
Uncaught (in promise) TypeError: ServiceWorker script at http://localhost:4321/qwik-prefetch-service-worker.js for scope http://localhost:4321/ encountered an error during installation. 
```

**Comment by thejackshelton** [comment]
> I'm running output: server in astro, i can't understand why this 404 message comes up in log? luckly, there's no problems running app realated this message.
> 
> ```
> "@builder.io/qwik": "^1.3.5"
> "@qwikdev/astro": "^0.4.0"
> "astro": "^4.1.2",
> 
> 14:40:20 [200] / 4ms
> 14:40:20 [200] /api/0 0ms
> 14:40:20 [404] /qwik-prefetch-service-worker.js
> ```

This is because of the new speculative module fetching, it does not break any pre-existing behavior. Aware of the issue, and reached out to the Astro team on how to fix it, since we need the file in the public folder when the user uses dev mode.

**Comment by PaulieScanlon** [comment]
Hey @thejackshelton, i'm experiencing the same issue. 

I see 404 on the service-worker locally, and an uncaught promise when i deploy. 

Here's a minimal reproduction repo if it's helpful.  [https://github.com/PaulieScanlon/reproduction-astro-qwik](https://github.com/PaulieScanlon/reproduction-astro-qwik)

... and here's a link to the bigger project i've been working on:

- Preview: [https://contentful-tags-data-viz.netlify.app/](https://contentful-tags-data-viz.netlify.app/)
- Repo: [https://github.com/PaulieScanlon/contentful-tags-data-viz](https://github.com/PaulieScanlon/contentful-tags-data-viz)

**Comment by PaulieScanlon** [comment]
@thejackshelton

 Boom! i updated the following packages and this has fixed the problem! 

Chalk another one up for "Action Jack" Shelton :) 

```
    "@builder.io/qwik": "^1.3.5",
    "@qwikdev/astro": "^0.4.3",
    "astro": "^4.1.3",
```

**Comment by TheElegantCoding** [comment]
I have the same error with this version

```json
{
    "@builder.io/qwik": "1.4.0",
    "@qwikdev/astro": "0.4.4",
    "astro": "4.2.1"
}
```

Reproduction [Repo](https://github.com/TheElegantCoding/astro_template_webpage)

**Comment by thejackshelton** [comment]
> I have the same error with this version
> 
> ```json
> {
>     "@builder.io/qwik": "1.4.0",
>     "@qwikdev/astro": "0.4.4",
>     "astro": "4.2.1"
> }
> ```
> 
> Reproduction [Repo](https://github.com/TheElegantCoding/astro_template_webpage)

Thanks for keeping us updated! You can ignore this error for now, the file shouldn't be in dev mode anyways.

**Comment by thejackshelton** [comment]
To update on this issue, here is my findings:

If we create a new `@qwikdev/astro` application, and run the dev server, then we do not get the 404 warning. In our application tab, we can see that no service workers are registed.

<img width="872" alt="image" src="https://github.com/QwikDev/astro/assets/104264123/60d4517b-55d0-4e19-8309-ebe01cb19d5c">

We can also see that the prefetch service worker and bundle graph scripts are not used in dev mode

<img width="1019" alt="image" src="https://github.com/QwikDev/astro/assets/104264123/0b3c844c-b07c-4b1c-ab30-0db8b2813914">

The interesting part, is when we build our application. A `qwik-prefetch-service-worker.js` script is created.

<img width="452" alt="image" src="https://github.com/QwikDev/astro/assets/104264123/a9610888-1005-4060-a9f3-3151355a8ede">

When we go back to dev mode, we can see that the service worker is active. It would appear that the service worker is activated in dev mode when it should not be.

I have a hunch here, that the problem lies in the fact that Astro uses the same port for dev mode and preview behavior. If we run `pnpm dev --port 5173` for example, we never get the 404 request, nor the service worker register.

In fact, if we go back to `4321`, and we unregister the service worker that seems to be created from our build in chrome's `application` tab, then we do not get the 404 in our terminal.

**Comment by thejackshelton** [comment]
Fixed in 0.4.8. Closing!
